### ISSUE TITLE
MacOS: starting the display in a thread errors

### ISSUE THREAD

**Issue by Neradoc** [FoamyGuy/Blinka_Displayio_PyGameDisplay#16]
Running on mac with a basic example I get this:
```
Exception in thread Thread-1:
Traceback (most recent call last):
  File "<path>/lib/python3.9/threading.py", line 973, in _bootstrap_inner
    self.run()
  File "<path>/lib/python3.9/threading.py", line 910, in run
    self._target(*self._args, **self._kwargs)
  File "<path>/blinka/.env/lib/python3.9/site-packages/blinka_displayio_pygamedisplay.py", line 115, in _pygame_refresh
    self._pygame_screen = pygame.display.set_mode(
pygame.error: NSWindow drag regions should only be invalidated on the Main Thread!

Assertion failed: (NSViewIsCurrentlyBuildingLayerTreeForDisplay() != currentlyBuildingLayerTree), function NSViewSetCurrentlyBuildingLayerTreeForDisplay, file NSView.m, line 13477.
Illegal instruction: 4
```
This seems to indicate that set_mode (or in general window init) must be done in the main thread.

I tried moving this section of code into the _initialize function before starting the thread, and that seemed to work so far:

https://github.com/FoamyGuy/Blinka_Displayio_PyGameDisplay/blob/28a344044768778360ac70227324a34f9beb187e/blinka_displayio_pygamedisplay.py#L103-L117

**Comment by mountainstorm** [comment]
Created a PR which fixed this: https://github.com/FoamyGuy/Blinka_Displayio_PyGameDisplay/pull/17

**Comment by FoamyGuy** [comment]
resolved by #17
